<a href="https://colab.research.google.com/github/nagaforjesus/python-projects/blob/main/Cancer_T_test_A_B_test.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

My Project focuses on Cancer Prevalence Across Age Groups.  
1. I will be performing a T-test -> Which is statistical test used to determine if there is a significant difference between the means of two groups.
2. Also, A/B test ->Traditionally A/B test is a statistical method used to compare two version of a product to determine which performs better. But we will use it for something more in this project.

N.B, the T-test and A/B test are related but not identical.

Expect results from this project.

* T-test: If the p-value is below 0.05, we reject the null hypothesis and conclude that there is a significant difference in cancer cases between the two age groups.
* A/B Test: Similarly, for the A/B test, a p-value below 0.05 would indicate a significant difference in cancer prevalence between males and females.


In [1]:
import pandas as pd

In [2]:
df = pd.read_csv('/content/Cancer_Prevalence_Age_Groups.csv')

# Display the first few rows of the dataset to check
df.head()

Age Group      Cancer Type  Gender Region  Cases
0     30-39  Prostate Cancer    Male   West    774
1     60-69    Breast Cancer    Male  South    942
2     60-69  Prostate Cancer  Female   East    440
3     18-29  Prostate Cancer  Female  South    763
4     70-80  Prostate Cancer  Female   West    295

In [8]:
# Filter data by the two age groups
age_18_29 = df[df['Age Group'] == '18-29']['Cases']
age_60_69 = df[df['Age Group'] == '60-69']['Cases']


In [9]:
from scipy import stats

# Perform the T-test
t_stat, p_value = stats.ttest_ind(age_18_29, age_60_69)

# Display the results
print(f"T-statistic: {t_stat}")
print(f"P-value: {p_value}")

# Interpretation
if p_value < 0.05:
    print("Result: Significant difference in cancer prevalence between the two age groups.")
else:
    print("Result: No significant difference in cancer prevalence between the two age groups.")


T-statistic: 0.9193920742074474
P-value: 0.39335128787047147
Result: No significant difference in cancer prevalence between the two age groups.


What the T-test found:

The test compared cancer cases in the two age groups: 18-29 and 60-69.
It calculated a number (T-statistic) to see how different these two groups are.
The p-value tells us if this difference is meaningful or just random.
The result:

The p-value is 0.393, which is higher than 0.05. This means the difference we see between the two groups isn’t big enough to be considered meaningful. It's likely just random variation.
What this means:

There’s no strong evidence that cancer cases are significantly different between these two age groups in this dataset.


In [10]:
# Filter data for males and females
males = df[df['Gender'] == 'Male']['Cases']
females = df[df['Gender'] == 'Female']['Cases']

# Perform the A/B test (Independent T-test)
from scipy import stats

t_stat, p_value = stats.ttest_ind(males, females)

# Print the results
print(f"T-statistic: {t_stat}")
print(f"P-value: {p_value}")

# Interpretation of the result
if p_value < 0.05:
    print("Result: Significant difference in cancer prevalence between males and females.")
else:
    print("Result: No significant difference in cancer prevalence between males and females.")


T-statistic: -0.9083760350088941
P-value: 0.37637760136913656
Result: No significant difference in cancer prevalence between males and females.


A/B Test Result Interpretation:
T-Statistic:

The T-statistic value is -0.908, which tells us how much the means of the two groups (males and females) differ in relation to the variation in the data.
The negative value simply indicates that the mean for males is lower than for females (but it's not statistically significant).
P-Value:

The p-value is 0.376, which is greater than the significance threshold of 0.05.
This means that the difference in cancer prevalence between males and females is not statistically significant.
Conclusion:

We fail to reject the null hypothesis, meaning there’s no evidence of a significant difference in cancer prevalence between males and females in this dataset.